### Creating input-target pairs

In this section we will implement a data loader that fetches the inut-target pairs using a sliding window approach

To start, we will first tokenize the whole "TEXT" using the BPE tokenizer

In [1]:
import tiktoken

In [2]:
tokenizer = tiktoken.get_encoding("gpt2")

In [3]:
with open("the-verdict.txt", "r", encoding="utf-8") as f:
    raw_text = f.read()


enc_text = tokenizer.encode(raw_text)
print(len(enc_text))

5145


The above text has 5145 token, after applying BPE tokenizer

In [5]:
enc_sample = enc_text[50:]

One of the easiste ways to create the input-target pairs for the nextword prediction task is to create 2 variables x & y

Where x contains input & y contains target, which are inputs shifted by 1